### PR TITLE
refactor: use ruff as the server

### PR DESCRIPTION
**Pull request by jfcherng**
## Proposal

As of `ruff` v0.4.0, it can be used as a LSP server. We no longer need the `ruff-lsp` Python wrapper. Plus, `ruff` is even faster than `ruff-lsp`, obviously.

ref: https://github.com/astral-sh/ruff/releases/tag/v0.4.0

## I am not going to keep tracking the latest `ruff` release in this PR. Remember to update the version number if you are going to merge this PR.

The current latest `ruff` release can be found on https://github.com/astral-sh/ruff/releases/latest

### CODE REVIEW

**Comment by jfcherng** [comment]
Technically, we no longer need to have Python installed on the machine now. We need it before because `ruff-lsp` is written in Python. 

`ruff` itself can be downloaded from GitHub release. That means `LSP-ruff` doesn't have to inherit `PipClientHandler`.

**Comment by rchl** [comment]
The changelog says that it's in alpha which makes me want to wait with switching to it since there might be potentially missing features or bugs to iron out.

**Comment by rchl** [comment]
BTW. We have github actions that can monitor github releases (used in LSP-rust-analyzer for example)

**Comment by jfcherng** [comment]
> BTW. We have github actions that can monitor github releases (used in LSP-rust-analyzer for example)

Thanks for the info. I am pretty sure that is what I was looking for repositories I forgot...

**Comment by LDAP** [comment]
I have one for LSP-clangd too https://github.com/sublimelsp/LSP-clangd/blob/main/.github/workflows/update_check.yaml

**Comment by jfcherng** [comment]
https://github.com/astral-sh/ruff/releases/tag/0.5.3

Ruff 0.5.3 marks the stable release of the Ruff language server and introduces revamped [documentation](https://docs.astral.sh/ruff/editors), including [setup guides for your editor of choice](https://docs.astral.sh/ruff/editors/setup) and [the language server itself](https://docs.astral.sh/ruff/editors/settings).

**Review by rchl — APPROVED** [review]

**Review by rchl — CHANGES_REQUESTED** [review]
I saw now that there are some setting changes: https://docs.astral.sh/ruff/editors/migration/

Will you handle that or do you want someone to take over?

**Comment by jfcherng** [comment]
> I saw now that there are some setting changes: [docs.astral.sh/ruff/editors/migration](https://docs.astral.sh/ruff/editors/migration/)
> 
> Will you handle that or do you want someone to take over?

done

**Review by rchl — APPROVED** [review]

File: LSP-ruff.sublime-settings [hunk 1]
```diff
@@ -3,22 +3,31 @@
...
         "globalSettings": {
-            // Custom arguments passed to ruff.
-            // See ruff documentation at https://github.com/charliermarsh/ruff/blob/main/README.md#configuration
-            "lint.args": [],
-            // Run Ruff on every keystroke (onType) or on save (onSave).
-            "lint.run": "onType",
+            // Path to a ruff.toml or pyproject.toml file to use for configuration.
+            // By default, Ruff will discover configuration for each project from the filesystem, mirroring the behavior of the Ruff CLI.
+            "configuration": null,
+            // The strategy to use when resolving settings across VS Code and the filesystem. By default, editor configuration is prioritized over ruff.toml and pyproject.toml files.
```
Comment (rchl) [inline LSP-ruff.sublime-settings:11]:
```suggestion
            // The strategy to use when resolving settings across editor and the filesystem. By default, editor configuration is prioritized over ruff.toml and pyproject.toml files.
```

File: sublime-package.json [hunk 2]
```diff
@@ -11,29 +11,77 @@
...
+                    "editorFirst",
+                    "filesystemFirst",
+                    "editorOnly"
+                  ],
+                  "enumDescriptions": [
+                    "Editor settings take priority over configuration files present in the workspace.",
+                    "Configuration files present in the workspace takes priority over editor settings.",
+                    "Ignore configuration files entirely i.e., only use editor settings."
+                  ],
+                  "markdownDescription": "The strategy to use when resolving settings across VS Code and the filesystem. By default, editor configuration is prioritized over `ruff.toml` and `pyproject.toml` files."
```
Comment (rchl) [inline sublime-package.json:32]:
```suggestion
                  "markdownDescription": "The strategy to use when resolving settings across editor and the filesystem. By default, editor configuration is prioritized over `ruff.toml` and `pyproject.toml` files."
```

**Review by rchl — APPROVED** [review]

**Pull request merged**